### ISSUE TITLE
[Feature Req.] Add ping graph on the status page

### ISSUE THREAD

**Issue by RisedSky** [louislam/uptime-kuma#637]
**Is it a duplicate question?**
No

**Is your feature request related to a problem? Please describe.**
I'm confused about the Ping graph, it is shown in the dashboard, but not in the status page next to the probe, this can be very useful

It can be in a checkbox, if the user want that feature, it checks it, and it's shown in the status page like this
![image](https://user-images.githubusercontent.com/23197999/136705708-c475ff71-f343-4466-a9f1-4c8c7f9d21b0.png)

**Comment by hypervtechnics** [comment]
I'd want to click on a monitor to see additional options. Then having all collapsed by default. I like the slim design where you can see quickly which sites are down.

**Comment by markdesilva** [comment]
On this subject, I would suggest an all-monitors-heartbeat graph on the dashboard where the average of all the heartbeats (100%, 97%, 99.2%, etc) vs time is shown. Gives a clear indication of all outages over the moving time period.

**Comment by RisedSky** [comment]
> I would suggest an all-monitors-heartbeat graph on the dashboard where the average of all the heartbeats (100%, 97%, 99.2%, etc) vs time is shown. Gives a clear indication of all outages over the moving time period.

I'm TOTALLY for THIS, this would be very cool !
Maybe an option to set it to be "30 days" history OR "all time" history ? 😄

**Comment by markdesilva** [comment]
> Not normal, you can set it in the settings in the dashboard for a default theme

Just checked again. It doesn't stick once I close the browser and open it again and log in. And across different browsers on different machines too, it always defaults to light mode.

**Comment by RisedSky** [comment]
> > Not normal, you can set it in the settings in the dashboard for a default theme
> 
> Just checked again. It doesn't stick once I close the browser and open it again and log in. And across different browsers on different machines too, it always defaults to light mode.

It's on the status page (go to the dashboard first): 
![image](https://user-images.githubusercontent.com/23197999/137412434-15e1319f-27ae-407e-9634-8b1dc1af2d29.png)

**Comment by markdesilva** [comment]
Hi, 

Thanks. 

For the status page its always dark theme, but the dashboard always reverts to light theme on different browsers and different machines and if I close the browser and reopen and log in again.

**Comment by RisedSky** [comment]
> Hi, 
> 
> Thanks. 
> 
> For the status page its always dark theme, but the dashboard always reverts to light theme on different browsers and different machines and if I close the browser and reopen and log in again.
> 
> 

This is due to the automatic option, set to dark in the dashboard on the settings tab

**Comment by grand-lotus-iroh** [comment]
Maybe, when you get a chance you can add this feature request to Project Plan 20 TODO or Ideas 💡 column?

https://github.com/louislam/uptime-kuma/projects/1

**Comment by UnclearGhostGT** [comment]
> I'd want to click on a monitor to see additional options. Then having all collapsed by default. I like the slim design where you can see quickly which sites are down.

I agree, having maybe a drop down button like an arrow down or something, and on click it opens the chart - or additional details - for that monitor would be great.



> > let me know if you want to go ahead with this, I can take a look 🙆🏻‍♂️
> 
> It's ok, but I would suggest to check the performance of Chart.js first. Something like would it be slow if there are 50 instances of Chart.js.

Would you be able to do like a load chart.js whenever the button is clicked to view the chart?

I'm thinking people wouldnt mind an extra few seconds to load the chart on the button click vs having all charts load making the entire page slow.

**Comment by RisedSky** [comment]
Will you work on this ? @louislam this is a good idea who gets a lot of thumbs up ☺️

**Comment by InsaneSplash** [comment]
I'm really excited to see the ping graphs on the customer public status page.

**Comment by bitsky6** [comment]
Any update on this?

**Comment by LoudSoftware** [comment]
Bump

**Comment by CommanderStorm** [comment]
@LoudSoftware @bitsky6 @RisedSky @InsaneSplash 

Please refrain from posting `+1` / requests for updates things on issues, as this makes issue-management harder.
Issues are for discussing what needs to be done how by whom.
[We use 👍🏻 on issues to prioritise work](https://github.com/louislam/uptime-kuma/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

As always: Pull requests are welcome; This is a community-driven project

**Comment by ehaughee** [comment]
I've got a really rough (simple) implementation of this feature and I'm curious if I should continue.

Here's what I've got so far
<img width="1349" alt="Screenshot 2023-07-07 at 10 32 06 PM" src="https://github.com/louislam/uptime-kuma/assets/312189/26c7bdc6-edaa-4953-97c6-1aa62f0266f3">

Thus far I've just added a `<PingChart ... />` to `PublicGroupList` and then added a `size` parameter, similar to the `HeartbeatBar`, to the `Pingchart`.  Based on the string value of `this.size` (basically just does it `=== "size"`) adjust or disable various chart elements/properties.

It seems like labeling or some other indicator of what the two UI elements (heartbeat and ping chart) are showing is necessary now that there's two and not just one.  Maybe some simple "Healthchecks" and "Response Time" labels but I'm not sure.

I *think* I'd like to add some min, avg, max stats to it.  Maybe 3 stacked values or AVG full height and then min/max stacked or something.  There's a lot of options with hover, or dropdowns/expansions, etc. but I really like the simplicity of this visual.  Maybe the row could just be clickable, if you're in admin mode, and that would take you to the monitor detail page for more info.  This kind of poses a question around who is this for, the service(s) owner or just the service(s) user.  There's definitely some overlap (particularly with my usecases) so more design is certainly warranted.  But even where it's at right now, I think it provides value as an at-a-glance look at whether a monitor's response time is out of the usual range. 

Anyway, my code needs to be cleaned up, made toggle-able, and tested, but I wanted to check on the direction before doing any additional work.

**Comment by CommanderStorm** [comment]
Reviewing walls of texts is always difficult.
If I missed something, feel free to ping.

In general, I have a preference for more, shorter pull requests, as this way changes are more focused and thus easier to follow.
This approach also allows the reviewer a better view into the contributing engeneers mind => hopefully reduces defects ^^

Conceptually, I like this addition, but have a few concerns:
- just adding this diagram next to the other one seems kind of counterintuitive.
  I think that adding this as an alternate (i.e. only displaying one diagram) might reduce visual clutter.
  Maybe showing the downtimes in this diagram could also be an option
- adding `min`, `avg`, `max` stats might be tricky to do without looking cluttered. I am curious about your implementation. (An option: lets split this into another PR => making this improvement eazier to review)
- different colour for threshold exceeding:
  - Given that we don't support his in the other views, I would definitely push this into another PR (keeping this one reviewable).
- making the row linkable should be another PR (getting merged quicker, easier to review), given that this is another issue
  - related issues: https://github.com/louislam/uptime-kuma/issues/2507 https://github.com/louislam/uptime-kuma/issues/2505 https://github.com/louislam/uptime-kuma/issues/1273

**Comment by ehaughee** [comment]
I'm definitely on board with multiple smaller PRs.  Been out of town for a bit but once I'm back I'll take a stab at adding a configuration value to swap to a ping graph here first. I am interested in your thoughts on how to show downtime in the chart.  Might be representable by just a lack of datapoints but I'd need to see how visually obvious that is.

**Comment by dumbasPL** [comment]
Looking at some other status pages out there, almost all the ones I've seen that had ping graphs had them in a separate section. This way they are way bigger, more readable, and also allow for adding extra features like the ability to select the time frame. 

Maybe we shouldn't try to cram an entire graph for every service but allow the user to pick a few services that will have their ping graphs visible in a separate section of the status page.

**Comment by bitsky6** [comment]
> Looking at some other status pages out there, almost all the ones I've seen that had ping graphs had them in a separate section. This way they are way bigger, more readable, and also allow for adding extra features like the ability to select the time frame.
> 
> Maybe we shouldn't try to cram an entire graph for every service but allow the user to pick a few services that will have their ping graphs visible in a separate section of the status page.

Why not just add it as a button/switch which will expand the current service with larger graph and more info if needed instead adding to all of them.

**Comment by ehaughee** [comment]
The more I think about this issue, the more I think the status page should just have service rows be links to their dashboard pages.  Any other option adds visual complexity (in the case of having the chart on each service) or just duplicates a portion of the dashboard (in the case of having an accordion/dropdown).

Not only is that implementation the simplest, but it's the same amount of clicks as a dropdown, has no performance implications for the page (don't have to worry about lazy-loading chartjs), and adds no visual complexity.

There does seem to be appetite for *some* sort of visibility into response time on the status page but I'm becoming more sure a chart on the status page is not the right way to provide that.

**Comment by tanglu1323123** [comment]
> I've got a really rough (simple) implementation of this feature and I'm curious if I should continue down my current path.
> 
> Here's what I've got so far <img alt="Screenshot 2023-07-07 at 10 32 06 PM" width="1349" src="https://user-images.githubusercontent.com/312189/251937741-26c7bdc6-edaa-4953-97c6-1aa62f0266f3.png">


Hi,
how can I set like yours ?
I really like and need that :)

**Comment by vincejv** [comment]
if anyone's still interested in this, just get telegraf + ping input or http input then visualize it with grafana, then you get your graph.
Use this grafana dashboard as a start https://grafana.com/grafana/dashboards/15225-ping-network-uptime/

**Comment by RisedSky** [comment]
> if anyone's still interested in this, just get telegraf + ping input or http input then visualize it with grafana, then you get your graph. Use this grafana dashboard as a start https://grafana.com/grafana/dashboards/15225-ping-network-uptime/

We shouldn't have this type of other program on the go.
We should instead have this directly in the product

**Comment by RyderCragie** [comment]
> Why not just add it as a button/switch which will expand the current service with larger graph and more info if needed instead adding to all of them.

I agree. Less to render = quicker load times. Only show it if the user requests it.

**Comment by kobayashi90** [comment]
Welp, how is this Feature req. sitting on the backseat, its basically what this statuspage system needs.
Please make it happen already.

**Comment by CommanderStorm** [comment]
> how is this Feature req. sitting on the backseat.

As with all other open-source projects, we rely on contributions. Our contribution guide [can be found here](https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md).
If that is not possible, you could consider supporting us at https://opencollective.com/uptime-kuma 

> Please make it happen already

This issue contains multiple smaller sub-issues. Some of them require a lot more changes than others.

What improvements are currently either planned or already in the next releases can be [found here](https://github.com/louislam/uptime-kuma/milestones).
Examples of shipping sub-features are the improvements to min, max, avg on charts in v2.0.
Note that a lot of maintainer effort is currently going into performance fixes ⇒ v2.0 will include a lot of effort on this front.

**Comment by dponzone** [comment]
+1 on that

Exploring Kuma, the first thing I noticed about status pages is this limitation.
If I want to use them to give access to a customer or a partner, they will be informed about the status of a service, but not the specific values.

**Comment by BelgianNoise** [comment]
I would honestly just love to see the 24h avg.
You could easily have a slider to choose whether to show it or not, so no performance issue for the ppl that dont care.

**Comment by kobayashi90** [comment]
Amazing to see that this still hasnt been merged, just 6 days until 4 years have past. Focus on actual features/bugfixes and unmess your 700~ issues. Hopefully one of the devs is reading this and improve this in next next coming Month. cheers.

**Comment by rtman** [comment]
Just curious if this is still on the roadmap and if there's any where to see status on this feature?

**Comment by CommanderStorm** [comment]
@kobayashi90 If you want to help declutter issues, I would be very open to that.

**Comment by CommanderStorm** [comment]
@rtman 

My previous comment seems to have gotten burried.

There’s no dedicated roadmap item for this feature at the moment - as with most parts of the project, it depends on someone from the community stepping up to implement it. The [milestones](https://github.com/louislam/uptime-kuma/milestones)
 only reflect what’s already planned or in active development.

If this feature is important to you, the best way to help move it forward is by contributing directly (see the [contribution guide](https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md)) or by supporting the project on [Open Collective](https://opencollective.com/uptime-kuma).
Anthother thing that you can do is this:
- #6260
